File: tools/templates/test/src/main/scala/test1/modules.scala
package org.tresamigos.smvtest.test1

import org.tresamigos.smv._

object M1 extends SmvModule("") with SmvOutput {
  override def requiresDS = Seq(input.table)

  override def run(i: runParams) = {
    i(input.table)
  }
}

object M2 extends SmvModule("") with SmvOutput {
  override def requiresDS = Seq(M1)

  override def run(i: runParams) = {
    i(M1)
  }
}

File: tools/templates/test/src/main/python/integration/test/test2/modules.py
import smv

from integration.test.test2 import input

class M1(smv.SmvModule):
    def requiresDS(self):
        return [input.table]

    def run(self, i):
        return i[input.table]

class M2(smv.SmvModule, smv.SmvOutput):
    def requiresDS(self):
        return [M1]

    def run(self, i):
        return i[M1]
